src/lib/selector/fast-auth-wallet.ts

```diff
@@ -0,0 +1,310 @@
+          receiverId as string,
+          actions.map(createAction),
+          localKey,
+        );
+        const transaction = nearAPI.transactions.createTransaction(
+          account.accountId,
+          nearAPI.utils.PublicKey.from(txAccessKey.public_key),
+          receiverId as string,
+          txAccessKey.access_key.nonce + 1,
+          actions.map(createAction),
+          nearAPI.utils.serialize.base_decode(block.header.hash),
+        );
+        const arg = {
+          transactions: [transaction],
+        };
+        const closeDialog = await _state.wallet.requestSignTransactions(arg);
+        const listener = (e: MessageEvent) => {
+          const shouldRemove = processSignedDelegates(params.relayerUrl, closeDialog, e);
+          if (shouldRemove) {
+            window.removeEventListener('message', listener, false);
+          }
+        };
+        window.addEventListener('message', listener);
+      } else {
+        const signedDelegate = await account.signedDelegate({
+          actions: actions.map((action) => createAction(action)),
+          blockHeightTtl: 60,
+          receiverId: receiverId as string,
+        });
+
+        await fetch(params.relayerUrl, {
+          method: 'POST',
+          mode: 'cors',
+          body: JSON.stringify(Array.from(encodeSignedDelegate(signedDelegate))),
+          headers: new Headers({ 'Content-Type': 'application/json' }),
+        });
+      }
+    },
+
+    async signAndSendTransactions({ transactions, callbackUrl }) {
+      const account = _state.wallet.account();
+      const { accessKey } = await account.findAccessKey('', []);
+
+      const needsFAK = transactions.some(({ receiverId }) => {
+        return accessKey.permission !== 'FullAccess' && accessKey.permission.FunctionCall.receiver_id !== receiverId;
+      });
+
+      if (needsFAK) {
+        const arg = {
+          transactions: await transformTransactions(transactions),
+          callbackUrl,
+        };
+        const closeDialog = await _state.wallet.requestSignTransactions(arg);
+        const listener = (e: MessageEvent) => {
+          const shouldRemove = processSignedDelegates(params.relayerUrl, closeDialog, e);
+          if (shouldRemove) {
+            window.removeEventListener('message', listener, false);
+          }
+        };
+        window.addEventListener('message', listener);
+      } else {
+        for (const { receiverId, signerId, actions } of transactions) {
+          const signedDelegate = await account.signedDelegate({
+            actions: actions.map((action) => createAction(action)),
+            blockHeightTtl: 60,
+            receiverId: receiverId as string,
+          });
+
+          await fetch(params.relayerUrl, {
+            method: 'POST',
+            mode: 'cors',
+            body: JSON.stringify(Array.from(encodeSignedDelegate(signedDelegate))),
+            headers: new Headers({ 'Content-Type': 'application/json' }),
+          });
```

Comment:
maybe not in this PR, but I see repeated code regards to sending to relayer, perhaps we could refactor this into a function and just pass body perhaps?